experimental/Schemes/src/ToricBlowups/attributes.jl

```diff
@@ -75,43 +103,45 @@ function center(bl::ToricBlowdownMorphism)
     I = ideal(S, x[j])
     II = ideal_sheaf(X, I)
     JJ = pushforward(bl, II)::IdealSheaf
-    bl.center = JJ
+    bl.center_unnormalized = JJ
   end
-  return bl.center
+  return bl.center_unnormalized
 end
 
 
 
 @doc raw"""
-    exceptional_divisor(bl::ToricBlowdownMorphism)
+    exceptional_prime_divisor(bl::ToricBlowdownMorphism)
 
-Return the exceptional divisor (as toric divisor)  of the toric blowdown morphism.
+Return the exceptional prime Weil divisor (as a toric divisor) of the ray used to construct the toric blowup. Note that this divisor need not be Cartier and this divisor need not coincide with the locus where the morphism is not an isomorphism.
 
 # Examples
 ```jldoctest
 julia> P3 = projective_space(NormalToricVariety, 3)
 Normal toric variety
 
-julia> blow_down_morphism = blow_up(P3, [0, 1, 1])
+julia> f = blow_up(P3, [0, 2, 3])
 Toric blowdown morphism
 
-julia> exceptional_divisor(blow_down_morphism)
-Torus-invariant, cartier, prime divisor on a normal toric variety
+julia> E = exceptional_prime_divisor(f)
+Torus-invariant, prime divisor on a normal toric variety
+
+julia> is_cartier(E)
+false
 ```
 """
-function exceptional_divisor(bl::ToricBlowdownMorphism)
-  if !isdefined(bl, :exceptional_divisor)
+function exceptional_prime_divisor(bl::ToricBlowdownMorphism)
+  if !isdefined(bl, :exceptional_prime_divisor)
     X = domain(bl)
     S = cox_ring(X)
     x = gens(S)
     j = index_of_new_ray(bl)
     help_list = [i == j ? 1 : 0 for i in 1:ngens(S)]
     td = toric_divisor(X, help_list)
-    @assert is_cartier(td) "exceptional divisor must be Cartier"
-    @assert is_prime(td) "exceptional divisor must be prime"
-    bl.exceptional_divisor = td
+    @assert is_prime(td) "exceptional prime divisor must be prime"
```

Comment:
This does not seem very helpful -- if it is an exceptional prime divisor by type, then it is forcibly prime and we do not need to assert it. If the assert is necessary, then the prime divisor type has a problem....